### PR TITLE
docs: create guide section

### PR DESCRIPTION
**Pull request by edeediong**

### CODE REVIEW

**Review by sourishkrout — APPROVED** [review]
👍 LGTM

File: docs/Guide/devcontainer.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+For engineers, the solution would be to document processes for handling incidents and automating everyday operations. As organizations seek to replace manual processes with executable code, runbooks have become more essential.
+
+[Runme](https://runme.dev/) is a tool that lets you run step-by-step processes, both manually and automatically, making it easier to handle and automate tasks. It's like a one-stop solution to speed up your development and operation processes.
+
+In this, we will give you a step-by-step guide on how to integrate Dev Container into your Runme and perform awesome actions.
+
+At the end of this guide you will learn how to:
+
+- Setup Runme in your code editor, Vscode
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:15]: VS Code

File: docs/Guide/devcontainer.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+## Introduction
+
+In the development process, the goal extends beyond having working and scalable software but simplifying routine tasks to reduce complexity.
+
+For engineers, the solution would be to document processes for handling incidents and automating everyday operations. As organizations seek to replace manual processes with executable code, runbooks have become more essential.
+
+[Runme](https://runme.dev/) is a tool that lets you run step-by-step processes, both manually and automatically, making it easier to handle and automate tasks. It's like a one-stop solution to speed up your development and operation processes.
+
+In this, we will give you a step-by-step guide on how to integrate Dev Container into your Runme and perform awesome actions.
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:11]: either "Dev Containers" or "a Dev Container". Also, "into your Runme notebook.

File: docs/Guide/devcontainer.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+For this guide, we will be using an already existing Python app. Feel free to use any application of your choice.
+
+1. Install Visual Studio Code
+
+If you haven’t installed this yet you should [install it now](https://code.visualstudio.com/).
+
+2. Install the Dev Containers extension ( formerly called *Remote- Container extension*)
+
+In your VS Code, navigate to the extension view by clicking on the **Extensions Icon** in the Activity bar on the side of the window. Search for **Dev Containers**  and install the extension.
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:33]: In VS Code

File: docs/Guide/devcontainer.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+![dev container installation](../../static/img/runme-devcontainer.png)
+
+3. In your Vs code project, create a Dev Container Configuration
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:37]: VS Code

File: docs/Guide/devcontainer.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+This will execute your Python app within the dev container.
+
+![python app](../../static/img/dev-container-python-ouput.png)
+
+**Note:**
+
+1. Ensure Docker is open and running on your system before you create a dev container.
+2. You do not need to create a separate dockerfile for this as the base image and settings needed for your development environment are specified in the dev container configuration (devcontainer.json). This eliminates the need for explicit Dockerfile maintenance and makes it simple to share and replicate the development environment across several machines.
+
+## **Setting up Runme on VSCode**
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:74]: VS Code

File: docs/Guide/devcontainer.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+Installing Runme in your VS Code is one of the various ways you can utilize the awesome features of Runme. To set up Runme on VS Code, follow our [installation guide.](https://docs.runme.dev/install#runme-for-vs-code)
+
+## **Running development tasks using Runme within your container**
+
+Runme works perfectly  to automate processes and conduct development operations within the container . To run your code with Runme in your dev container, follow the steps below:
+
+- Create and open your **READ.me** file in your VScode.
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:82]: README.md? VS Code.

File: docs/Guide/devcontainer.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+![remote icon](../../static/img/remote-container-icon.png)
+
+Now a new view opens that prompts you to select and open a remote window.
+
+- Click on **Dev container** or **Add dev container configuration files**
+- Choose a dev container configuration based on your programming language. This guide uses a Python app, so the Python dev container will be used.
+- Now choose the version you want and add any other features you want to install.
+- Now click **OK**. Your container configuration gets added, and the container starts.
+
+You will notice that your READ.me file switches over to a normal markdown setting. You will need to take it back to a Runme setting to enable you to run your code with Runme in a dev container. To do this,
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:95]: README.md? Markdown (capitalized)

File: docs/Guide/devcontainer.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+![set shebang](../../static/img/runme-devcontainer-shebang.png)
+
+- Now click the **Run** icon and view your output on the Runme terminal.
+
+![ouput](../../static/img/output-devcontainer.png)
+
+## **Other Features**
+
+Some other actions that can be performed in your Dev container include cloning repository, configuring container features, exploring a volume, etc. The image below shows you a non-exhaustive list of some of these actions.
```
Comment (joseeantonior) [inline docs/Guide/devcontainer.md:113]: Either Dev Container or dev container.

File: docs/Guide/terraform.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+1. **Cloud Provider :**
+
+   Ensure that you have an AWS account (for the sake of this article we will be using AWS) and access key credentials. If you don't have an AWS account, you can create one [here](https://aws.amazon.com/).
+
+2. **Terraform Installation:**
+
+   Install Terraform on your local machine. You can download the latest version from the [official Terraform website.](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+
+3. Runme Installation:
```
Comment (joseeantonior) [inline docs/Guide/terraform.md:35]: Let's use the same format as in the two points above.

File: docs/Guide/terraform.md [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+   Install runme on your local machine. You can install runme from the [official page](https://docs.runme.dev/install)
```
Comment (joseeantonior) [inline docs/Guide/terraform.md:36]: Runme

File: docs/Guide/terraform.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+
+here is an example of how to create an ec2 with terraform via runme
```
Comment (joseeantonior) [inline docs/Guide/terraform.md:38]: EC2

File: docs/Guide/terraform.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+output:
+
+![terraform](../../static/img/terraform-apply.png)
+
+Terraform will prompt you to confirm the changes. Type **yes** and press Enter.
+
+1. **Verify EC2 Instance:**
+
```
Comment (joseeantonior) [inline docs/Guide/terraform.md:110]: This extra line is not displaying.

Comment (joseeantonior) [reply docs/Guide/terraform.md:110]: If we have a 1 here, we should have a 2 below. Otherwise, remove the 1.

**Review by joseeantonior — COMMENTED** [review]
We should address the comments I listed.

**Pull request merged**